agent/api/appnet/client_linux.go

```diff
@@ -41,6 +42,9 @@ func (cl *client) GetStats(config serviceconnect.RuntimeConfig) (map[string]*pro
 	if err != nil {
 		return nil, err
 	}
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.Wrapf(err, "received non-OK HTTP status %v from Service Connect stats endpoint", resp.StatusCode)
```

Comment:
Is it safe to return before `defer resp.Body.Close()` call?

Reply:
Also should we do add the response body to the error? It might contain useful info?

Reply:
good callout. Will change it